src/rgw/rgw_op.cc

```diff
@@ -1928,8 +1928,14 @@ void RGWCreateBucket::execute()
   if (!store->is_meta_master()) {
     JSONParser jp;
     op_ret = forward_request_to_master(s, NULL, store, in_data, &jp);
-    if (op_ret < 0)
+    if (op_ret < 0) {
+      if (op_ret == -EINVAL){
+	// Taking the easy way out, basically some issue in processing request from our side
+	// return a 5XX instead of an -EINVAL to the end user
+	op_ret = -ERR_SERVICE_UNAVAILABLE;
+      }
```

Comment:
returning 503 seems reasonable to me. but we should probably move this up into `forward_request_to_master()` so that the other callers (`RGWSetBucketVersioning` and `RGWDeleteBucket`) do the same